rust/private/rustc.bzl

```diff
@@ -754,18 +748,24 @@ def _compute_rpaths(toolchain, output_dir, dep_info):
     Returns:
         depset: A set of relative paths from the output directory to each dependency
     """
-    if not dep_info.transitive_dylibs:
+    preferreds = [get_preferred_artifact(lib) for lib in dep_info.transitive_noncrates.to_list()]
+
+    # TODO(augie): I don't understand why this can't just be filtering on
+    # _is_dylib(lib), but doing that causes failures on darwin and windows
```

Comment:
Or you mean that _is_dylib finds dylibs on mac/windows, and the other thing doesn't, and therefore with _is_dylib we trigger `fail` on line 760?

Reply:
Yes. It's super weird, but given that doing it this way (which seems wrong) prevents any regressions I'm willing to live with that for now in the name of making progress.

Reply:
(Assuming you are, anyway.)

Reply:
Ok, this PR is already touching many sensitive pieces, let's defer fixing this into a followup PR.